exercises/leap/example_gen.go

```diff
@@ -15,7 +16,8 @@ func main() {
 		log.Fatal(err)
 	}
 	var j js
-	if err := gen.Gen("leap.json", &j, t); err != nil {
+	f := path.Join("exercises", "leap", "canonical-data.json")
```

Comment:
I totally missed this issue, I stumbled upon this through trying to implement a test generator in another exercise.
I was thinking the same thing, I'll look into cancelling my other PR and updating the `Gen` function. Leave this one unmerged and hopefully I can update it along with the others.